src/main/java/courseManager/InstanceList.java

```diff
@@ -0,0 +1,143 @@
+import java.util.ArrayList;
+import java.util.Objects;
+
+public class InstanceList {
+    private ArrayList<AssessmentInstance> listOfAssessmentInstances;
+
+    private String pluralTitle;
+    private String singularTitle;
+    private int totalNumberOfInstances;
+    private int numberOfCommittedInstances;
+    private int numberOfSubmittedInstances;
+
+    /**
+     * Create a new InstanceList object with a list of AssessmentInstances
+     * @param pluralTitle the plural title of the assessment
+     *              e.g. "Quizzes", "Midterm", "Homework Assignments"
+     * @param totalNumberOfInstances the total number of instances of the assessment
+     */
+
+    public InstanceList(String pluralTitle, int totalNumberOfInstances) {
+        this.listOfAssessmentInstances = new ArrayList<>();
+        this.singularTitle = toSingular(pluralTitle);
+        for (int i = 0, j = 1; i < totalNumberOfInstances; i++, j++){
+            this.addInstance(this.singularTitle + " #" + j);
+        }
+    }
+
+    public String toSingular(String pluralTitle){
+        if (pluralTitle == "Quizzes"){
+            return "Quiz";
+        }
+        if (pluralTitle.charAt(pluralTitle.length()-1) == 's'){
+            return singularTitle.substring(0, pluralTitle.length()-1);
+        }
+        return singularTitle;
+    }
+
+    public int getTotalNumberOfInstances() {
+        return totalNumberOfInstances;
+    }
+
+    public int getNumberOfCommittedInstances() {
+        int numberOfCommittedInstances = 0;
+        for (AssessmentInstance instance : listOfAssessmentInstances) {
+            if (instance.isCommitted()) {
+                numberOfCommittedInstances++;
+            }
+        }
+        return numberOfCommittedInstances;
+    }
+
+    public int getNumberOfSubmittedInstances() {
+        int numberOfSubmittedInstances = 0;
+        for (AssessmentInstance instance : listOfAssessmentInstances) {
+            if (instance.isSubmitted()) {
+                numberOfSubmittedInstances++;
+            }
+        }
+        return numberOfSubmittedInstances;
+    }
+
+    public double[] getAllMarks() {
+        double[] allMarks = new double[totalNumberOfInstances];
+        for (int i = 0; i < totalNumberOfInstances; i++) {
+            allMarks[i] = listOfAssessmentInstances.get(i).getMark();
+        }
+        return allMarks;
+    }
+
+
+    public double[] getCommittedMarks() {
+        double[] committedMarks = new double[getNumberOfCommittedInstances()];
+        int i = 0;
+        for (AssessmentInstance instance : listOfAssessmentInstances) {
+            if (instance.isCommitted()) {
+                committedMarks[i] = instance.getMark();
+                i++;
+            }
+        }
+        return committedMarks;
+    }
+
+    public void addInstance(String name) {
+        AssessmentInstance instance = new AssessmentInstance(name);
+        listOfAssessmentInstances.add(instance);
+        totalNumberOfInstances++;
```

Comment:
As already mentioned, check to see if we've gone above the total before adding an instance.

Reply:
Done!